client/js/components/prevent-overlapping.js

```diff
@@ -0,0 +1,42 @@
+import {nodeList, featureTest} from '../util';
+
+const preventOverlapping = (els) => {
+  if (!featureTest('position', 'sticky')) {
+    return;
+  }
+  const elements = nodeList(els).reverse();
+  const hideOverlapping = () => {
+    function getElementMeta() {
+      for (let i = 0; i < elements.length; i++) {
+        let elementMeta = {
+          currentEl: elements[i],
+          nextDomEl: elements[i - 1],
+          currentElHeight: elements[i].offsetHeight,
+          currentElCssTop: parseInt(window.getComputedStyle(elements[i], null).getPropertyValue('top'), 10),
+          currentElActualTop: elements[i].getBoundingClientRect().top,
+          nextDomElActualTop: elements[i - 1] ? elements[i - 1].getBoundingClientRect().top : null,
+          prevDomEls: elements.slice(i + 1)
+        };
+        if (elementMeta.currentElActualTop === elementMeta.currentElCssTop) {
+          return elementMeta;
+        }
+      }
+    }
+
+    const elementMeta = getElementMeta();
+    if (elementMeta !== undefined) {
+      elementMeta.prevDomEls.forEach((e) => {
+        e.style.opacity = 0;
+      });
+      if (elementMeta.nextDomElActualTop <= elementMeta.currentElHeight + elementMeta.currentElCssTop) {
+        elementMeta.currentEl.style.opacity = 0;
+      } else {
+        elementMeta.currentEl.style.opacity = 1;
+      }
+    }
+  };
```

Comment:
This is a lot of work going on on scroll - it's going to be a massive performance hog...